src/Deck.java

```diff
@@ -0,0 +1,542 @@
+      Deck doubleDeck = new Deck(2);
+      
+      int i = 0;
+      while (i < doubleDeck.cards.length)
+      {
+         System.out.println( doubleDeck.cards[i].toString() );
+         i += 1;
+      }
+      System.out.println("----------- 2 Decks (Shuffled) --------------------");
+      
+      i = 0;
+      doubleDeck.init(2);
+      doubleDeck.shuffle();
+      
+      while (i < doubleDeck.cards.length)
+      {
+         System.out.println( doubleDeck.cards[i].toString() );
+         i += 1;
+      }
+      System.out.println("-------------------------------------------------\n");
+      
+      System.out.println("----------- 1 Deck (Unshuffled) -------------------");
+      
+      Deck singleDeck = new Deck();
+      
+      i = 0;
+      
+      while (i < singleDeck.cards.length)
+      {
+         System.out.println( singleDeck.cards[i].toString() );
+         i += 1;
+      }
+      System.out.println("----------- 1 Deck (Shuffled) ---------------------");
+      
+      i = 0;
+      singleDeck.init(1);
+      singleDeck.shuffle();
+      
+      while (i < singleDeck.cards.length)
+      {
+         System.out.println( singleDeck.cards[i].toString() );
+         i += 1;
+      }
+      System.out.println("-------------------------------------------------\n");
+      
+   }
+   
+
+   public final int MAX_CARDS = 312;
+   private static Card[] masterPack = new Card[52];
+   private Card[] cards;
+   private int topCard;
+   
+   // Constructors
+   public Deck(int numPacks)
+   {  
+      allocateMasterPack();
+      init(numPacks);
+      topCard = 0;
+   }
+   
+   public Deck()
+   {
+      allocateMasterPack();
+      cards = masterPack;
+   }
+   
+   // Re-populates the cards[] with new cards. Resetting it to its original,
+   //unshuffled state.
+   // Methods
+   public void init(int numPacks)
+   {
+      cards = new Card[numPacks * 52];
+      int index = 0;
+      
+      while (numPacks > 0)
+      {
+         for (int i = 0; i < masterPack.length; i++)
+         {
+            cards[index] = masterPack[i];
+            index++;
+         }
+         
+         numPacks -= 1;
+      }     
+   }
+  
+   // Shuffles the cards[] by iterating through the cards[] and placing the
+   // card in each index in a random index.
+   public void shuffle()
+   {
+      Random rand = new Random();
+      int num;
+      Card tempCard;
+      
+      for (int i = 0; i < cards.length; i++)
+      {
+         num = rand.nextInt(52);
+         tempCard = cards[num];
+         cards[num] = cards[i];
+         cards[i] = tempCard;
+      }
+   }
+   
+   // Returns the card at the 'topCard' position of the deck and simulates the
+   // removal from the deck by incrementing topCard by 1. Also, throws an
+   // exception if the 'topCard' index exceeds the number of cards in the deck.
+   public Card dealCard()
+   {
+      if (topCard == cards.length)
+      {
+         throw new NullPointerException("NO MORE CARDS LEFT TO DEAL!");
```

Comment:
Null pointer exception is not what you want here. This 
there is also no catch for this throw.
A better check